nodes/text/export_gcode.py

```diff
@@ -0,0 +1,311 @@
+        row = col.row(align=True)
+        row.scale_y = 4.0
+        row.operator(TEXT_IO_CALLBACK, text='Export Gcode').fn_name = 'process'
+
+    def update_socket(self, context):
+        self.update()
+
+    def process(self):
+        # manage data
+        feed = self.feed
+        feed_v = self.feed_vertical
+        feed_h = self.feed_horizontal
+        layer = self.layer_height
+        layer = self.inputs['Layer Height'].sv_get()
+        vertices = self.inputs['Vertices'].sv_get()
+        flow_mult = self.inputs['Flow Mult'].sv_get()
+
+        # data matching
+        vertices = list_of_lists(vertices)
+        flow_mult = list_of_lists(flow_mult)
+        layer = list_of_lists(layer)
+        vertices, flow_mult, layer = match_longest_lists([vertices, flow_mult, layer])
+        print(vertices)
+        print(layer)
+
+        # open file
+        if self.folder == '':
+            folder = '//' + os.path.splitext(bpy.path.basename(bpy.context.blend_data.filepath))[0]
+        else:
+            folder = self.folder
+        if '.gcode' not in folder: folder += '.gcode'
+        path = bpy.path.abspath(folder)
+        file = open(path, 'w')
+        #file.write('M92 E' + format(self.esteps, '.4f') + '\n')
+        try:
+            for line in bpy.data.texts[self.start_code].lines:
+                file.write(line.body + '\n')
+        except:
+            pass
+
+        # sort vertices
+        if self.auto_sort and self.gcode_mode == 'RETR':
+            sorted_verts = []
+            for curve in vertices:
+                # mean z
+                listz = [v[2] for v in curve]
+                meanz = mean(listz)
+                # store curve and meanz
+                sorted_verts.append((curve, meanz))
+            vertices = [data[0] for data in sorted(sorted_verts, key=lambda height: height[1])]
+
+        # initialize variables
+        e = 0
+        last_vert = mathutils.Vector((0,0,0))
+        maxz = 0
+        path_length = 0
+
+        printed_verts = []
+        printed_edges = []
+        travel_verts = []
+        travel_edges = []
+
+        # write movements
+        for i in range(len(vertices)):
+            curve = vertices[i]
+            first_id = len(printed_verts)
+            for j in range(len(curve)):
+                v = curve[j]
+                v_flow_mult = flow_mult[i][j]
+                v_layer = layer[i][j]
+
+                # record max z
+                maxz = max(maxz,v[2])
+
+
+                # first point of the gcode
+                if i ==  j == 0:
+                    printed_verts.append(v)
+                    file.write('G92 E0 \n')
+                    params = v[:3] + (feed,)
+                    to_write = 'G1 X{0:.4f} Y{1:.4f} Z{2:.4f} F{3:.0f}\n'.format(*params)
+                    file.write(to_write)
+                    #file.write('G1 X' + format(v[0], '.4f') + ' Y' + format(v[1], '.4f') + ' Z' + format(v[2], '.4f') + ' F' + format(feed, '.0f') + '\n')
```

Comment:
take a leap of faith and remove these commented out lines, the replacement code is fine.